lib/puppet/provider/pulp_role/api.rb

```diff
@@ -0,0 +1,113 @@
+
+require File.expand_path('../../../util/pulp_util', __FILE__)
+
+Puppet::Type.type(:pulp_role).provide(:api) do
+  commands :pulp_admin => '/usr/bin/pulp-admin'
+
+  mk_resource_methods
+
+  def initialize(resource={})
+    super(resource)
+    @pulp = Puppet::Util::PulpUtil.new
+  end
+
+  def exists?
+    r = @pulp.get_role_info(name())
+    !r.nil?()
+  end
+
+  def create
+    manage_role('create', name())
+  end
+
+  def destroy
+    manage_role('delete', name())
+  end
+
+  def users
+    users = @pulp.get_role_users(name())
+    if users.nil?
+      nil
```

Comment:
Should this be a ```return nil```?

Reply:
Yes, I'm not used to Ruby at all, but I thought ``nil`` ``return nil`` and ``return`` were the same thing? Is ``return nil`` a better style?
I just tried roles with empty user array and empty permissions hash, it works well with both syntaxes.

Reply:
The difference is that ```return nil``` results in ```@users``` not being set. Both return nil in the end. In the current code this if-block does nothing so it can either be removed or changed to ```return nil```. Not sure which is the better way.

Reply:
You are right, those lines are of no use. I deleted them in cef1604 , thanks!